include/aws/s3/private/s3_auto_ranged_get.h

```diff
@@ -53,6 +53,12 @@ struct aws_s3_auto_ranged_get {
     } synced_data;
 
     uint32_t initial_message_has_range_header : 1;
+
+    bool initial_message_has_start_range;
```

Comment:
Using bool because we can not have a pointer to bit field https://stackoverflow.com/questions/13547352/c-cannot-take-address-of-bit-field.

Reply:
Thanks, I have updated it to option 1 for now. I would have preferred option 2 but don't want to include that refactor in this PR.